win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1713,8 +1713,194 @@ UINT FASTCALL IntGetCharSet(INT nIndex, FT_ULong CodePageRange1)
+                   ft_face->charmaps[i]->platform_id, ft_face->charmaps[i]->encoding_id);
+
+            switch (ft_face->charmaps[i]->platform_id)
+            {
+                default:
+                    cmap_def = ft_face->charmaps[i];
+                    break;
+                case 0: /* Apple Unicode */
+                    cmap0 = ft_face->charmaps[i];
+                    break;
+                case 1: /* Macintosh */
+                    cmap1 = ft_face->charmaps[i];
+                    break;
+                case 2: /* ISO */
+                    cmap2 = ft_face->charmaps[i];
+                    break;
+                case 3: /* Microsoft */
+                    cmap3 = ft_face->charmaps[i];
+                    break;
+            }
+        }
+#ifdef __REACTOS__
+// Wine bug: for each loop iteration, a FT_Set_Charmap() call is made.
+// ReactOS fix: Instead, loop until we retrieve all the cmap pointers
+// of interest, then do one single FT_Set_Charmap() call using the
+// preferred cmap.
+    }{
+#endif
+
+        if (cmap3) /* prefer Microsoft cmap table */
+            ft_err = FT_Set_Charmap(ft_face, cmap3);
+        else if (cmap1)
+            ft_err = FT_Set_Charmap(ft_face, cmap1);
+        else if (cmap2)
+            ft_err = FT_Set_Charmap(ft_face, cmap2);
+        else if (cmap0)
+            ft_err = FT_Set_Charmap(ft_face, cmap0);
+        else if (cmap_def)
+            ft_err = FT_Set_Charmap(ft_face, cmap_def);
+    }
+
+    return ft_err == FT_Err_Ok;
+}
+
+/* Borrowed from Wine */
+static FT_Encoding pick_charmap( FT_Face face, int charset )
+{
+    static const FT_Encoding regular_order[] = { FT_ENCODING_UNICODE, FT_ENCODING_APPLE_ROMAN, FT_ENCODING_MS_SYMBOL, 0 };
+    static const FT_Encoding symbol_order[]  = { FT_ENCODING_MS_SYMBOL, FT_ENCODING_UNICODE, FT_ENCODING_APPLE_ROMAN, 0 };
+    const FT_Encoding *encs = regular_order;
+
+    if (charset == SYMBOL_CHARSET) encs = symbol_order;
+
+    while (*encs != 0)
+    {
+        if (select_charmap( face, *encs )) break;
+        encs++;
+    }
+
+    if (!face->charmap && face->num_charmaps)
+    {
+        if (!FT_Set_Charmap(face, face->charmaps[0]))
+            return face->charmap->encoding;
+    }
+
+    return *encs;
+}
+
+#if 0
+//
+// NOTE: This was the old charmap selection code in TextIntUpdateSize().
+//
+static
+void
+FindFallbackCharmap(
+    _In_ FT_Face face)
+{
+    FT_CharMap charmap, found = NULL;
+    INT error, n;
+
+    if (face->charmap)
+        return;
+
+    DPRINT1("WARNING: No charmap selected!\n");
+    DPRINT1("This font face has %d charmaps\n", face->num_charmaps);
+__debugbreak();
+
+    /*
+     * Selects a unicode charmap, if any.
+     * Prefers one that supports UCS-4, otherwise use UCS-2.
+     * See also FreeType's src/base/ftobjs.c!find_unicode_charmap().
+     */
+    // error = FT_Select_Charmap(face, FT_ENCODING_UNICODE);
+    for (n = 0; n < face->num_charmaps; n++)
+    {
+        //
+        // Specific 
```

Comment:
Extra trailing space
```suggestion
        // Specific
```